# Plan
This model is to help us set a bar on how well our other models should perform

In [23]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
# find a model to import (random forest)
from sklearn.model_selection import TimeSeriesSplit
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

In [21]:
weather = pd.read_csv('weatherdataCMSE.csv')
weather = weather.rename(columns={'STATION': 'Station',
                                  'NAME': 'Name',
                                  'DATE': 'Date',
                                  'AWND': 'Avg Wind Speed',
                                  'PGTM': 'Peak Gust Time',
                                  'PRCP': 'Precipitation',
                                  'TAVG': 'Avg Temp',
                                  'TMAX': 'Max Temp',
                                  'TMIN': 'Min Temp',
                                  'WDF2': 'Direction of Fastest 2 Min Wind',
                                  'WDF5': 'Direction of Fastest 5 Sec Wind',
                                  'WSF2': 'Fastest 2 Min Wind Speed',
                                  'WSF5': 'Fastest 5 Sec Wind Speed',
                                  'WT01': 'Fog, Ice Fog, or Freezing Fog',
                                  'WT02': 'Heavy Fog, Heavy Freezing Fog',
                                  'WT03': 'Thunder',
                                  'WT08': 'Smoke or Haze'})
columns_to_keep = ['Date', 'Precipitation']
weather = weather[columns_to_keep]

shifted_precip = weather['Precipitation'].shift(1)
weather['Precipitation Yesterday'] = shifted_precip
weather = weather.drop(index=weather.index[0])
weather = weather.dropna()
for column in weather.columns:
    missing = weather[column].isnull().sum()
    print(f"{column}: {(missing/len(weather))*100:.3f}%")

Date: 0.000%
Precipitation: 0.000%
Precipitation Yesterday: 0.000%


In [34]:
mae = mean_absolute_error(y_true=weather['Precipitation'], y_pred=weather['Precipitation Yesterday'])
mse  = mean_squared_error(y_true=weather['Precipitation'], y_pred=weather['Precipitation Yesterday'])
me   = np.mean(weather['Precipitation'] - weather['Precipitation Yesterday'])

print(f"Mean Absolute Error Score: {mae:.3f}")
print(f"ME   (bias): {me:.3f}")
print(f"RMSE: {np.sqrt(mse):.3f}")
print(f"MSE: {mse:.3f}")

Mean Absolute Error Score: 0.135
ME   (bias): 0.000
RMSE: 0.334
MSE: 0.112


# Base Prediction Errors

>**Mean Absolute Error**: 0.135
>
>**Mean Error**: 0.000
>
>**Mean Squared Error**: 0.112




### <u>**Definitions**<u>
* Mean Absolute Error: Measures the average size of your mistakes. On average, the baseline model is off by 0.135 inches of precipitation.
* Mean Error: A mean error of 0 means no strong average bias. A score greater than 0 means underpredicting and a score less than 0 means overpredicting.
* Root Mean Squared Error: Larger errors are penalized more because it's squared but the square root function takes it back to the original units.
